integration_tests/auth/test_auth.py

```diff
@@ -124,6 +130,83 @@ def test_cannot_log_in_with_deleted_user(
+        self,
+        role_or_user: _RoleOrUser,
+        _get_user: _GetUser,
+        _passwords: Iterator[_Password],
+        _smtpd: smtpdfix.AuthController,
+        _smtp_config: SmtpConfig,
+    ) -> None:
+        u = _get_user(role_or_user)
+        logged_in_user = u.log_in()
+        logged_in_user.create_api_key()
+        _initiate_password_reset(u.email)
+        assert len(_smtpd.messages) == 1
+        msg: Message = _smtpd._messages.pop()
+        assert msg["to"] == u.email
+        assert msg["from"] == _smtp_config.mail_from
+        token = _extract_password_reset_token(msg)
+        new_password = next(_passwords)
+        assert new_password != u.password
+        _reset_password(token, new_password)
+        with _EXPECTATION_401:
+            # old password should no longer work
+            u.log_in()
+        with _EXPECTATION_401:
+            # old tokens should no longer work
+            logged_in_user.create_api_key()
+        newer_password = next(_passwords)
+        assert newer_password != new_password
+        with _EXPECTATION_401:
+            # reset token can only work once
+            _reset_password(token, newer_password)
+        # new password should work
+        new_profile = replace(u.profile, password=new_password)
+        replace(u, profile=new_profile).log_in().create_api_key()
+
+    # @pytest.mark.parametrize("role_or_user", [_MEMBER, _ADMIN])
+    # def test_deleted_user_should_not_receive_email_after_initiating_password_reset(
+    #     self,
+    #     role_or_user: _RoleOrUser,
+    #     _get_user: _GetUser,
+    #     _passwords: Iterator[_Password],
+    #     _smtpd: smtpdfix.AuthController,
+    #     _smtp_config: SmtpConfig,
+    # ) -> None:
+    #     u = _get_user(role_or_user)
+    #     logged_in_user = u.log_in()
+    #     logged_in_user.create_api_key()
+    #     _DEFAULT_ADMIN.delete_users(u)
+    #     _initiate_password_reset(u.email)
+    #
+    # @pytest.mark.parametrize("role_or_user", [_MEMBER, _ADMIN])
+    # def test_deleted_user_should_not_be_able_to_reset_password_using_token_from_email(
+    #     self,
+    #     role_or_user: _RoleOrUser,
+    #     _get_user: _GetUser,
+    #     _passwords: Iterator[_Password],
+    #     _smtpd: smtpdfix.AuthController,
+    #     _smtp_config: SmtpConfig,
+    # ) -> None:
+    #     u = _get_user(role_or_user)
+    #     logged_in_user = u.log_in()
+    #     logged_in_user.create_api_key()
+    #     _initiate_password_reset(u.email)
+    #     assert len(_smtpd.messages) == 1
+    #     msg: Message = _smtpd._messages.pop()
+    #     assert msg["to"] == u.email
+    #     assert msg["from"] == _smtp_config.mail_from
+    #     token = _extract_password_reset_token(msg)
+    #     new_password = next(_passwords)
+    #     assert new_password != u.password
+    #     _DEFAULT_ADMIN.delete_users(u)
+    #     with _EXPECTATION_401:
+    #         _reset_password(token, new_password)
```

Comment:
Did you mean to add these?

Reply:
Yes, these are legitimate tests, but for some reason they would never halt when the `pytest-smtpd` fixture is added, and when the fixture is not added, the tests are totally fine, so it's a mystery.